src/main/java/org/terasology/structureTemplates/events/SpawnStructureEvent.java

```diff
@@ -15,15 +15,15 @@
  */
 package org.terasology.structureTemplates.events;
 
-import org.terasology.entitySystem.event.Event;
+import org.terasology.entitySystem.event.AbstractConsumableEvent;
 import org.terasology.structureTemplates.components.SpawnBlockRegionsComponent;
 import org.terasology.structureTemplates.util.transform.BlockRegionTransform;
 
 /**
  * When sent to entities with components like {@link SpawnBlockRegionsComponent} then the structure described by
  * that entity will be generated.
  */
-public class SpawnStructureEvent implements Event {
+public class SpawnStructureEvent extends AbstractConsumableEvent {
```

Comment:
I don't think SpawnStructureEvent should be a consumable event. A lot of components are just adding to the spawned structure.

Maybe instead of getting the blocks from SpawnBlockRegion, it GrowStructureComponent could define it's own regions?